### DISCUSSION TITLE
How can I use a .env on my renative project?

### DISCUSSION THREAD

**Discussion by mendoza** [pavjacko/renative#1107] [General]
Hi everyone!
I'm working on a TV app right now, and I want to be able to use .env files for certain configurations (like the base url for the api),
I tried using a field on my runtime config on my `renative.json` with `{{env.BASE_URL}}` however these only works with env variables set on the command line. After that, I tried using `react-native-dotenv` or anything like that, didn't really affected the `process.env` on the React code. 

Is there a specific way I haven't seen?

**Comment by mendoza** [comment]
I found the `renative.private.json` file that could be used instead of .env files, however I would like to also know if it's possible to use them.

**Comment by pauliusguzas** [comment]
@mendoza hello, thanks for this question, have you tried using [dotenv](https://www.npmjs.com/package/dotenv?activeTab=readme) package? I know we have used it with great success

**Reply by mendoza** [reply]
I tried running it on tv (or in native really, I get some error about crypto, fs and other libraries dotenv uses)

**Reply by pauliusguzas** [reply]
@mendoza Can you create a ticket with described issue, reproduction sequence and logs? thank you

**Reply by mendoza** [reply]
Sure let me get you that, before I do, where would you call the dotenv.config? I think my questions would boildown to, what would be the entry point for dotenv

**Comment by battistini01** [comment]
I resolved using Next env. Create your .env.local file and put your vars with NEXT_PUBLIC_ prefix for each

**Reply by jonathandion** [reply]
This solution works only when the platform is `web`.
